src/screens/WebsocketResearchScreen/SampleChatScreen.tsx

```diff
@@ -57,45 +63,106 @@ const SampleChatScreen = () => {
     sendChat,
     updateChatContinuity
   } = useChatScreenHook(ANONYMOUS);
+  const {replyPreview, clearReplyPreview} = useMessageHook();
+  const {moveToSignedChannel} = useMoveChatTypeHook();
 
   const updatedChats = updateChatContinuity(chats);
+  const [loading, setLoading] = React.useState(false);
+  const {anonProfileId} = useProfileHook();
+  const ownerChat = selectedChannel?.rawJson?.channel?.members?.find(
+    (item: any) => item.user_id === anonProfileId
+  );
+  const memberChat = selectedChannel?.rawJson?.channel?.members?.find(
+    (item: any) => item.user_id !== anonProfileId
+  );
 
   const renderChatItem = React.useCallback(({item, index}) => {
     return <BaseChatItem type={ANONYMOUS} item={item} index={index} />;
   }, []);
 
+  const betterSocialMember = selectedChannel?.rawJson?.better_channel_member;
+
   const scrollToEnd = () => {
     flatlistRef.current?.scrollToEnd();
   };
 
+  const moveChatSigned = async () => {
+    try {
+      setLoading(true);
+      await moveToSignedChannel({
+        oldChannelId: selectedChannel?.id,
+        targetUserId: memberChat.user_id
+      });
+    } catch (e) {
+      console.log('error moving chat to signed channel', e);
+    } finally {
+      setLoading(false);
+    }
+  };
+  React.useEffect(() => {
+    return () => {
+      clearReplyPreview();
+    };
+  }, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of hooks and state variables seems appropriate. However, in the `moveChatSigned` function, there's a console.log statement for error logging. It would be better to replace this with a more robust error handling mechanism or logging system.

```diff
-      console.log('error moving chat to signed channel', e);
+      // Replace with your preferred logging system
+      Logger.error('error moving chat to signed channel', e);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->